app/services/organizations/update_invoice_numbering_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Organizations
+  class UpdateInvoiceNumberingService < BaseService
+    Result = BaseResult[:organization]
+
+    def initialize(organization:, document_numbering:)
+      @organization = organization
+      @document_numbering = document_numbering
+      super
+    end
+
+    def call
+      result.organization = organization
+
+      return result if organization.document_numbering == document_numbering
+
+      if organization.per_customer? && document_numbering == "per_organization"
+        last_invoice = organization.invoices.non_self_billed.with_generated_number.order(created_at: :desc).first
```

Comment:
there is one edge scenario here we reproduce in QA, it is actually not affecting document numbering.

with `per_organization` creates invoices in `draft`:

`invoice_1`
`invoice_2`
`invoice_3`

then finalize `invoice_3` -> this one gets `organization_sequential_id = 1`

switch to `per_customer`

then finalize the other invoices, both get `organization_sequential_id = 0`

switch back to `per_organization`.

here the last invoice created is `invoice_3`, hence its `organization_sequential_id` is overwritten and changes from `1` to `3`.

The next invoice finalized will get a good `organization_sequential_id` and the invoice number will be fine.